lib/reducers/adapterReducer.js

```diff
@@ -57,6 +57,8 @@ const DEFAULT_ADAPTER_STATUS = 'Select serial port';
 const ImmutableRoot = Record({
     api: { adapters: [], selectedAdapter: null },
     adapters: List(),
+    ports: List(),
+    versionInfos: List(),
```

Comment:
The serialPortReducer.js in core keeps track of the list of ports, including selected port. Can be accessed through `getState().core.serialPort.selectedPort`. So I do not think we need the list of `ports` here as well.

I understand the need to keep versionInfo here, because of `openAdapter(port, versionInfo)` in adapterActions.js. It should not be necessary to keep multiple versionInfos though. I would suggest setting `state.versionInfo` for the selected adapter in `adapterOpened()`, and setting it back to null in `closeAdapter()`.

Reply:
Oh, I see. I tried to find it in app but it was not there. I will update this one.